### PR TITLE
Decklist rework

### PR DESCRIPTION
**Pull request by NIK-VOLO**

### CODE REVIEW

**Comment by NIK-VOLO** [comment]
This is just progress toward the new decklist. 

Still needs:
- Fix card selection 
- Decklist card styles
- Text scaling issues for mobile
- stuff

File: app/components/cardList.tsx [hunk 1]
```diff
@@ -68,15 +67,20 @@ const isNumberInCardValue = (search: string, value: string) => {
...
   const [cardList, setCardList] = useState(props.cardArray);
 
   useEffect(() => {
     setCardList(props.cardArray);
-  },[]);
+  }, []);
 
   const updateCollectionCount = (isIncremented: boolean, cardNum: string) => {
-    var cardToUpdateIndex = cardList.findIndex(card => card.cardNum === cardNum)
+    var cardToUpdateIndex = cardList.findIndex(
```
Comment (NIK-VOLO) [inline app/components/cardList.tsx:81]: Is it possible to do an O(1) lookup here?

File: app/components/providers/deckListProvider/DeckListProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    if (deckListLength >= deckListLimit) {
+      Swal.fire({
+        title: "<strong>Card Limit Reached</strong>",
+        html: `<b>Can only have ${deckListLimit} cards in your deck.</b>`,
+        icon: "error",
+        confirmButtonColor: "#f27474",
+        confirmButtonText: "Close",
+      });
+      return; // BEN NOTE: MIGHT UPDATE THIS LATER TO SHOW SWAL IF CHANGING FROM BYEAH PRIME
```
Comment (NIK-VOLO) [inline app/components/providers/deckListProvider/DeckListProvider.tsx:85]: Your comment seems like an important edge case to handle!

File: app/components/providers/deckListProvider/DeckListProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            );
+        } else {
+          copyDeckList[i].count--;
+        }
+        setDeckListLength(deckListLength - 1);
+        break;
+      }
+    }
+    setDeckList([...copyDeckList]);
+  };
```
Comment (NIK-VOLO) [inline app/components/providers/deckListProvider/DeckListProvider.tsx:156]:
```suggestion
  const removeFromDeckList = (cardNum: string) => {
  if (!cardsData || !deckList) {
    return;
  }

  const cardObj = cardsData[parseInt(cardNum) - 1];

  // Find the index of the card in the deckList
  const cardIndex = deckList.findIndex((deckCard) => deckCard.cardNum === cardNum);

  // Return early if the card is not found in the deckList
  if (cardIndex === -1) {
    return;
  }

  const copyDeckList = [...deckList];
  if (copyDeckList[cardIndex].count <= 1) {
    copyDeckList.splice(cardIndex, 1);

    // Remove the card from the epicArray when its count is 0
    if (cardObj.type[0] === "✦") {
      setEpicArray((prevEpicArray) => prevEpicArray.filter((currCardNum) => currCardNum !== cardNum));
    }
  } else {
    copyDeckList[cardIndex].count--;
  }

  setDeckListLength((prevDeckListLength) => prevDeckListLength - 1);
  setDeckList(copyDeckList);
};

```

**Review by Benjamin-Hayduchok — APPROVED** [review]
yahooo

**Pull request merged**